pkgs/test/tool/host.dart

```diff
@@ -216,36 +207,34 @@ StreamChannel<dynamic> _connectToIframe(String url, int id) {
     if (js_util.getProperty(location, 'href') != iframe.src) return;
 
     message.stopPropagation();
+    windowSubscription.cancel();
 
     if (message.data['ready'] == true) {
       // This message indicates that the iframe is actively listening for
       // events, so the message channel's second port can now be transferred.
-      channel.port2.start();
+      var channel = dom.createMessageChannel();
+      _domSubscriptions[id] = dom.Subscription(channel.port1, 'message',
```

Comment:
Should we assert that there isn't one already just in case?

Reply:
I don't think we ever compile and run this script with assertions enabled since we manually compile it. We could potentially explore recompiling with asserts during CI at some point in the future so it won't hurt to add them.

Reply:
We could do a regular check/throw also I suppose? But I am not sure we would actually want to crash. It does indicate an unexpected situation and would result in an uncancelled listener though.

Reply:
I'll probably leave this code path using assert for now.

We can discuss more whether to us non-assert checks for the new code path when I have that implementation ready.